server/src/main/java/org/elasticsearch/common/geo/GeometryParser.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.geo.builders.ShapeBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.geo.geometry.Geometry;
+import org.elasticsearch.geo.utils.WellKnownText;
+import org.elasticsearch.index.mapper.BaseGeoShapeFieldMapper;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+/**
+ * An utility class with a geometry parser methods supporting different shape representation formats
+ */
+public final class GeometryParser {
+
+    private GeometryParser() {
+
+    }
+
+    /**
+     * Parses supplied XContent into Geometry
+     */
+    static Geometry parse(XContentParser parser, BaseGeoShapeFieldMapper shapeMapper) throws IOException, ParseException {
```

Comment:
It would be great to remove the dependency on a field mapper here. maybe require the three attributes explicitly